File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    target = priorities[location]
    answer = 0
    idx = 0
    
    maxList = list(set(priorities))
    maxList.sort()
    maxList.reverse()
    
    count = 0
    maxVal = maxList[count]
    
    while count < 10:
        if maxVal == target:
            newIdx = idx
            for i in range(len(priorities)):
                if newIdx >= len(priorities):
                    newIdx = 0
                if priorities[newIdx] == maxVal:
                    if newIdx == location:
                        return answer+1
                    answer+=1
                newIdx+=1
        newIdx = idx
        for i in range(len(priorities)):
            if newIdx >= len(priorities):
                newIdx = 0
            if priorities[newIdx] == maxVal:
                idx = newIdx
                answer+=1
            newIdx+=1
        count+=1
        maxVal = maxList[count]

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
import java.io.*;
class Solution {
    public int solution(int[] priorities, int location) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> {
            return b[0] - a[0];
        });
        Queue<int[]> queue = new LinkedList<>(); 
        for(int i = 0; i< priorities.length ; i++){
            pq.add(new int[] {priorities[i], i});
            queue.add(new int[] {priorities[i], i});
        }
        
        int count = 0;
        while(!queue.isEmpty()){
            int[] first = queue.poll();
            if(first[0] == pq.peek()[0]){
                pq.poll();
                count++;
                if(first[1] == location) return count;
            }else{
                queue.add(first);
            }
        }
        return count;
    }
}